sympy/core/sympify.py

```diff
@@ -266,6 +266,21 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     except AttributeError:
         pass
 
+    #Support for basic numpy datatypes
+    if type(a).__module__ == 'numpy':
+        import numpy as np
+        if np.isscalar(a):
+            try:
+                return sympify(np.asscalar(a))
+            except RuntimeError:
+                from sympy.core import Float
+                a = str(list(np.reshape(np.asarray(a),
+                                (1, np.size(a)))[0]))[1:-1]
+                return Float(a)
```

Comment:
The numpy float is converted to it's string representation, and then to sympy Float, hence no precision is lost.The precision is exactly same as the input numpy float.
At first I tried using a = str(a) , but that led to precision loss
`k = np.float128(1.1234567891234568)` ----> `str(k)` ---> `'1.12345678912'`.

With the above translation : 
```
>>> k = np.float128(1.1234567891234568)
>>> k
1.1234567891234568116
>>> sympify(k)
1.1234567891234568116
>>> sympify(k) == k
True
```